app/routes/onboarding/index.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { Form, useSubmit, useActionData } from '@remix-run/react';
+import { ActionFunction, json, redirect } from '@remix-run/node';
+import debounce from 'lodash.debounce';
+import axios from 'axios';
+import TimezoneSelect from 'react-timezone-select';
+import UserInput from '../../components/common/userInput';
+import Dropdown from '../../components/common/dropdown';
+import { Button } from '../../components/Button';
+
+// import { isUsernameAvailable } from '~/models/username.server';
+
+interface UserFormInterface {
+  username: string;
+  firstname: string;
+  lastname: string;
+  timezone: string;
+}
+
+interface UserDetailsInterface {
+  apiHost: string;
+  authToken: string;
+  page: number;
+  formTitlesAndSubtitles: Array<object>;
+}
+
+const isUsernameAvailable = async (host: string, token: string, username: string) => {
+  const url = `${host}/api/v1/users/usernameCheck/${username}`;
+  try {
+    const response = await axios.get(url, {
+      headers: {
+        // Authorization: `Bearer ${token}`,
+        'Content-Type': 'application/json',
+      },
+      withCredentials: true,
+    });
+    return response.data.data.available;
+  } catch (error) {
+    console.error('err', error);
+    return error;
+  }
+};
+
+export const action: ActionFunction = async ({ request, params }) => {
+  const formData = Object.fromEntries(await request.formData());
+
+  const url = `${process.env.API_HOST}/api/v1/users/self`;
+  const { username, firstname, lastname, timezone } = formData;
+
+  console.log(formData);
+
+  const errors = {
```

Comment:
We have used this in multiple places. Good to extract

Reply:
removed from one place